dandiapi/api/models/zarr.py

```diff
@@ -296,7 +308,20 @@ class Meta(BaseZarrArchive.Meta):
             models.UniqueConstraint(
                 name='unique-dandiset-name',
                 fields=['dandiset', 'name'],
-            )
+            ),
+            # Requires importing the django `Length` function
+            models.CheckConstraint(
+                name='nonempty-checksum',
+                check=models.Q(checksum__length__gt=0),
```

Comment:
What scenario does this account for?

Reply:
It just ensures that the field can't be empty :shrug:.

Realistically it doesn't do much as there's lots of invalid values between the empty string and a valid checksum, but it just felt natural to add. Do you think it should be removed?

Reply:
I think it should be removed. After looking at it, it's a particularly confusing check because `select length(null::text) > 0` evaluates to `null` (passing the check). That means this check is functionally equivalent to `~Q(checksum='')`. We should probably add a RegexValidator and maybe even check constraint for that, but I think we should punt given how close this PR is to the finish line.